src/WebJobs.Extensions.Http/HttpTriggerAttributeBindingProvider.cs

```diff
@@ -290,8 +299,11 @@ internal static void ApplyBindingData(object target, IDictionary<string, object>
                 var bindingData = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
                 if (request.ContentLength != null && request.ContentLength > 0)
                 {
-                    string body = await request.ReadAsStringAsync();
-                    Utility.ApplyBindingData(body, bindingData);
+                    if (string.IsNullOrEmpty(request.ContentType) || string.Equals(request.ContentType, "application/json", StringComparison.OrdinalIgnoreCase))
```

Comment:
This is the main change. Avoid the body read + json deserialization when we can. E.g. if the content type is specified as text/html, we shouldn't be trying to json deserialize it.